wgengine/magicsock/magicsock.go

```diff
@@ -2367,6 +2374,15 @@ func (c *Conn) SetDERPHeader(header http.Header) {
 	c.derpHeader.Store(&header)
 }
 
+func (c *Conn) SetDERPRegionDialer(dialer func(ctx context.Context, region *tailcfg.DERPRegion) net.Conn) {
+	c.derpRegionDialer.Store(&dialer)
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	for _, d := range c.activeDerp {
```

Comment:
Is there any chance `c.activeDerp` could contain a result where `d.c` is the `c *Conn` itself? Thus leading to a mutex deadlock?

Reply:
Or maybe `d.c` refers to client, in which case there should be no issue.

Reply:
`d.c` refers to client, so should be good I believe!